### ISSUE TITLE
[BUG] glitch in test 02_02_TestPtpSimpleLoopbackMasterRecord16000Hz32b32b4ch

### ISSUE THREAD

**Issue by wszypelt** [thesofproject/sof#4080]
**Describe the bug**
Glitch was found in test 02_02_TestPtpSimpleLoopbackMasterRecord16000Hz32b32b4ch
Playback pipeline is connected to slave SSP, capture pipeline is connected to master SSP.

**Topology**
```
pipe_plb
+------------------------------+
|-------+   +------+   +-----+ |   +----+
|| Host +-->+Buffer+-->+ Dai +---->+SSP1|
|-------+   +------+   +-----+ |   +----+
+------------------------------+    |  |
                                    |  |
pipe_cap                            |  |
+------------------------------+    |  |
|-------+   +------+   +-----+ |   +----+
|| Host +<--+Buffer+<--+ Dai +---->+SSP0|
|-------+   +------+   +-----+ |   +----+
+------------------------------+

```
**To Reproduce**
Run test with Diagnostic driver 02_02_TestPtpSimpleLoopbackMasterRecord16000Hz32b32b4ch
With options --playback-iterations=75

**Reproduction Rate**
1/75

**Environment**
1) Branch name and commit hash of the 2 repositories: sof (firmware/topology) and linux (kernel driver).
    * SOF BRANCH: main HASH: dc0ab8a449d6aea470f787a8aac8067d9c547ec2
2) Name of the platform(s) on which the bug is observed.
Platform: ICL RVP
3) Test environment:
BRANCH: master HASH: 180819001ee5c678ee2b6a53b4931c605a14e89c

**Screenshots or console output**
![image](https://user-images.githubusercontent.com/82375849/115731494-7bed6500-a387-11eb-8b2d-66e9a8cc30d0.png)

[02_02_~1.ZIP](https://github.com/thesofproject/sof/files/6359051/02_02_.1.ZIP)

**Comment by lgirdwood** [comment]
@wszypelt @zrombel Not sure if we can use the same buffer for both host and DAI DMA - please confirm if topology diagram is correct.

**Comment by wszypelt** [comment]
The same issue in the same reproduction appears in test
02_01_TestPtpSimpleLoopbackMasterPlayback16000Hz32b32b4ch

**Comment by lgirdwood** [comment]
@zrombel @kkarask @wszypelt @XiaoyunWu6666 @keqiaozhang @keyonjie @slawblauciak fwiw  we are seeing a lot of glitches being reported recently in various **loopback** tests.
Can we do the following here

1)  Align these under one issue and mark the others as duplicates, they all seem to show repeated or skipped samples which is usually indicative of races between DMA and DSP buffer IO.
2) Duplicate these failures using an existing upstream topology as its difficult to reproduce for developers who do not have access to HW with the slim test framework.

**Comment by lgirdwood** [comment]
Duplicate of simpler https://github.com/thesofproject/sof/issues/4076

**Comment by kkarask** [comment]
Glitch was found also in test where playback pipeline is connected to master SSP, capture pipeline is connected to slave SSP (02_01_TestPtpSimpleLoopbackMasterPlayback48000Hz32b32b4ch).
Reproduction rate: 2/75.

![image](https://user-images.githubusercontent.com/82518823/126306238-519b41a7-4bc4-448d-a65a-1b7dc5b552a4.png)

**Comment by lgirdwood** [comment]
@kkarask we think we may be getting closer to why the python driver and the no_codec CI devices are showing some issues as more reproducible.  Python driver and no codec mode are probably sending IPCs for some sequences faster that the kernel driver in codec mode (as codec mode means lots of slow & blocking I2C IO with codecs).

Can you rerun the test with #4471 locally merged and let us know if there are any changes to above result with Python driver.
@mwasko fyi.

**Comment by kkarask** [comment]
@lgirdwood I rerun tests on #4471  and tests passed, but problem can still exists because reproduction rate is not high. See also related issue: https://github.com/thesofproject/sof/issues/4180

**Comment by lgirdwood** [comment]
> @lgirdwood I rerun tests on #4471 and tests passed, but problem can still exists because reproduction rate is not high. See also related issue: #4180

Thanks @kkarask, do you mean that #4471 decreases the reproduction rate ? if so by how much ?

**Comment by kkarask** [comment]
@lgirdwood yes, it looks like reproduction rate was decreased. After run 3000 tests result is 0.001% fail.

**Comment by lgirdwood** [comment]
@lyakh fyi
